build.py

```diff
@@ -425,18 +425,23 @@ def handle_build_node(self, n):
 
             # Strip out the functions from shared library builds.
             if '$SHLINK' in str(n.executor):
-                assert len(lines) == 3
+                # Linux or Windows
```

Comment:
I think I'd break this into something like
```python
if myEnv.ToolchainIs('msvc'):
    assert len(lines) == 5
else:
    assert len(lines) == 3
```

But its also possible that this block will need to be split up more once you finish the work for DLLs.